src/js/controller/instream-adapter.js

```diff
@@ -123,6 +125,32 @@ var InstreamAdapter = function(_controller, _model, _view, _mediaPool) {
         return this;
     };
 
+
+    this.enableAdsMode = function(clickThroughUrl) {
+        if (_inited || _destroyed) {
+            return;
+        }
+        this.addAdProgramTimeListener();
+        this.applyProviderListeners(_instreamProvider);
+        this.on(INSTREAM_CLICK, () => {
+            if (_model.get('state') === STATE_PAUSED) {
+                _controller.playVideo();
+            } else if (clickThroughUrl) {
+                _controller.trigger(AD_CLICK, { clickThroughUrl });
+                _controller.pause();
+                window.open(clickThroughUrl);
+            }
+        });
+        return this;
+    };
+
+    this.updateTimeInAdsMode = function(event) {
```

Comment:
where is this being called? If DAI creates the provider instance it can trigger these events.